pallets/dynamic-evm-base-fee/src/benchmarking.rs

```diff
@@ -0,0 +1,75 @@
+// This file is part of Astar.
+
+// Copyright (C) 2019-2023 Stake Technologies Pte.Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// Astar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Astar is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Astar. If not, see <http://www.gnu.org/licenses/>.
+
+use super::*;
+
+use frame_benchmarking::v2::*;
+use frame_support::traits::Hooks;
+use frame_system::RawOrigin;
+
+#[benchmarks]
+mod benchmarks {
+    use super::*;
+
+    #[benchmark]
+    fn base_fee_per_gas_adjustment() {
+        let (first_block, second_block) = (T::BlockNumber::from(1u32), T::BlockNumber::from(2u32));
+        let init_bfpg = BaseFeePerGas::<T>::get();
+
+        // Setup actions, should ensure default value is written to storage.
+        Pallet::<T>::on_initialize(first_block);
+        Pallet::<T>::on_finalize(first_block);
+        Pallet::<T>::on_initialize(second_block);
```

Comment:
Maybe add an assertion here to check against the default value?

Reply:
It's not so much important that value is equal to the default one, but that the key|value exists.
Probably a minor optimization, but I assume if we read a value that doesn't exist in storage (yet), it's a bit cheaper to do it since no decoding is required - we just use the default one.

Probably an overkill 😅.
I will add a check that value exists in storage 👍